pkg/server/plugin/upstreamauthority/vault/vault_test.go

```diff
@@ -88,29 +89,33 @@ func (vps *VaultPluginSuite) Test_Configure() {
 			name:       "Configure plugin with AppRole authentication params given as environment variables",
 			configTmpl: testAppRoleAuthConfigWithEnvTpl,
 			envKeyVal: map[string]string{
-				"VAULT_APPROLE_ID":        "test-approle-id",
+				"VAULT_APPROLE_ID":        "test-approle-idd",
 				"VAULT_APPROLE_SECRET_ID": "test-approle-secret-id",
 			},
 		},
-	}
-
-	for _, c := range cases {
-		for k, v := range c.envKeyVal {
-			os.Setenv(k, v)
-		}
-
-		p := vps.newPlugin()
-
-		req := vps.getTestConfigureRequest(fmt.Sprintf("https://%v/", addr), c.configTmpl)
-
-		ctx := context.Background()
-		_, err = p.Configure(ctx, req)
-		vps.Require().NoError(err)
-		vps.Require().NotNil(p.vc.vaultClient.Token())
-
-		for k := range c.envKeyVal {
-			os.Unsetenv(k)
-		}
+		{
+			name:       "Multiple authentication methods configured",
+			configTmpl: testMultipleAuthConfigsTpl,
+			err:        "only one authentication method can be configured",
+		},
+	} {
+		c := c
+		vps.Run(c.name, func() {
+			for k, v := range c.envKeyVal {
+				os.Setenv(k, v)
+				defer os.Unsetenv(k)
```

Comment:
IMO: calling 'defer' in loop like this has a risk of resource leaks in general.
In this case it's a test, so is it ok?

Reply:
Yes, in general calling defer from a loop is considered bad practice. I doubt it would be a problem here but it would certainly "smell". Perhaps a better solution is to put something like this before the for loop:
```
defer func() {
    for k := range c.envKeyVal {
        os.Unsetenv(k)
    }
}()
for k, v := ...
```